### ISSUE TITLE
leftIconConfig OnClick error and rightIconConfig Widget Error 

### ISSUE THREAD

**Issue by AndreLuizNogueira** [SimformSolutionsPvtLtd/flutter_calendar_view#432]
I Just Tested Version 1.3.0 and found 2 possible errors.

leftIconConfig: IconDataConfig(
  icon: (context) => Icon(Icons.arrow_back_ios, color: Colors.white),
),
rightIconConfig: IconDataConfig(
  icon: (context) =>
      Icon(Icons.arrow_forward_ios, color: Colors.white),
),

1 - leftIconConfig: When you set this property, onclick is disabled;

2 - rightIconConfig. This property is not changing the icon. It is always using the default icon.

I went back to 1.2.0 and leftIcon and rightIcon worked fine;

leftIcon: const Icon(Icons.arrow_back_ios, color: Colors.white),
rightIcon: const Icon(Icons.arrow_forward_ios, color: Colors.white),

**Comment by shubham-jitiya-simform** [comment]
Hi @AndreLuizNogueira 
Thanks for reporting this issue.
`1 - leftIconConfig: When you set this property, onclick is disabled;` - This is the expected behaviour you can check [here](https://pub.dev/documentation/calendar_view/latest/calendar_view/IconDataConfig/icon.html) in the documentation.
`2 - rightIconConfig. This property is not changing the icon. It is always using the default icon.` - This will be fixed in PR #437.

### Example
Refer this snippet to provide change default icon
   ```
  headerStyle: HeaderStyle(
        leftIconConfig: IconDataConfig(
          icon: (_) => InkWell(
            onTap: () => debugPrint('On left icon tapped'),
            child: Padding(
              padding: EdgeInsets.all(16),
              child: Icon(Icons.arrow_left, color: Colors.red),
            ),
          ),
        ),
        rightIconConfig: IconDataConfig(
          icon: (_) => InkWell(
            onTap: () => debugPrint('On right icon tapped'),
            child: Padding(
              padding: EdgeInsets.all(16),
              child: Icon(Icons.arrow_right, color: Colors.red),
            ),
          ),
        ),
      )
```

### Preview
<img width="271" alt="Screenshot 2024-12-30 at 11 56 20 AM" src="https://github.com/user-attachments/assets/f0376a26-0a2a-4ca8-9337-e52e0c8478d4" />

**Comment by AndreLuizNogueira** [comment]
> Hi @AndreLuizNogueira 
> Thanks for reporting this issue.
> `1 - leftIconConfig: When you set this property, onclick is disabled;` - This is the expected behaviour you can check [here](https://pub.dev/documentation/calendar_view/latest/calendar_view/IconDataConfig/icon.html) in the documentation.
> `2 - rightIconConfig. This property is not changing the icon. It is always using the default icon.` - This will be fixed in PR #437.
> 
> ### Example
> Refer this snippet to provide change default icon
>    ```
>   headerStyle: HeaderStyle(
>         leftIconConfig: IconDataConfig(
>           icon: (_) => InkWell(
>             onTap: () => debugPrint('On left icon tapped'),
>             child: Padding(
>               padding: EdgeInsets.all(16),
>               child: Icon(Icons.arrow_left, color: Colors.red),
>             ),
>           ),
>         ),
>         rightIconConfig: IconDataConfig(
>           icon: (_) => InkWell(
>             onTap: () => debugPrint('On right icon tapped'),
>             child: Padding(
>               padding: EdgeInsets.all(16),
>               child: Icon(Icons.arrow_right, color: Colors.red),
>             ),
>           ),
>         ),
>       )
> ```
> 
> ### Preview
> <img width="271" alt="Screenshot 2024-12-30 at 11 56 20 AM" src="https://github.com/user-attachments/assets/f0376a26-0a2a-4ca8-9337-e52e0c8478d4" />
> 

Hello. Sorry about the first one, I just migrated from version 1.2.0 without testing, where all I had was the widget customization. I believe this deserves a breaking change notice, because I was using version 1.3.0 for months without ontap working, I only noticed the problem recently because I migrated to material 3 and had to check everywhere.
